src/vs/workbench/contrib/notebook/browser/services/notebookKernelHistoryServiceImpl.ts

```diff
@@ -79,12 +82,28 @@ export class NotebookKernelHistoryService extends Disposable implements INoteboo
 
 		if (notEmpty) {
 			const serialized = this._serialize();
-			this._storageService.store(NotebookKernelHistoryService.STORAGE_KEY, JSON.stringify(serialized), StorageScope.WORKSPACE, StorageTarget.MACHINE);
+			this._storageService.store(NotebookKernelHistoryService.STORAGE_KEY, JSON.stringify(serialized), StorageScope.WORKSPACE, StorageTarget.USER);
 		} else {
 			this._storageService.remove(NotebookKernelHistoryService.STORAGE_KEY, StorageScope.WORKSPACE);
 		}
 	}
 
+	private _restoreState(): void {
+		const serialized = this._storageService.get(NotebookKernelHistoryService.STORAGE_KEY, StorageScope.WORKSPACE);
+		if (serialized) {
+			try {
+				for (const [viewType, kernels] of JSON.parse(serialized)) {
+					const linkedMap = this._mostRecentKernelsMap[viewType] ?? new LinkedMap<string, string>();
+					for (const entry of kernels.entries) {
+						linkedMap.set(entry, entry, Touch.AsOld);
+					}
```

Comment:
Should `linkedMap` get saved back to `this._mostRecentKernelsMap`?

Reply:
good catch!